lib/puppet/http/redirector.rb

```diff
@@ -56,6 +56,9 @@ def redirect_to(request, response, redirects)
         next if header.casecmp('Authorization').zero? && request.uri.host.casecmp(location.host) != 0
         next if header.casecmp('Cookie').zero? && request.uri.host.casecmp(location.host) != 0
       end
+      # Allow Net::HTTP to set its own Accept-Encoding header to avoid errors with HTTP compression.
+      # See https://github.com/puppetlabs/puppet/issues/9143
+      next if header.casecmp('Accept-Encoding')
```

Comment:
If `casecmp` returns `0`, then it's interpreted as truthy. I think you want:

```suggestion
      next if header.casecmp('Accept-Encoding').zero?
```